recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,756 @@
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+
+
+    def _patch_source(self):
+        # Note that VTK's cmake will insert its own CMAKE_MODULE_PATH at the
+        # front of the list.  This is ok as long as there is nothing in that
+        # path that will be found before our conan cmake files...
+        # That is why we delete the third party finders.
+        # Else, we have to patch VTK's CMakeLists.txt, like so:
+        #####
+        # diff --git a/CMakeLists.txt b/CMakeLists.txt
+        # index c15890cfdc..022f704d75 100644
+        # --- a/CMakeLists.txt
+        # +++ b/CMakeLists.txt
+        # @@ -7,7 +7,7 @@ if (POLICY CMP0127)
+        #  endif ()
+        #
+        #  set(vtk_cmake_dir "${VTK_SOURCE_DIR}/CMake")
+        # -list(INSERT CMAKE_MODULE_PATH 0 "${vtk_cmake_dir}")
+        # +list(APPEND CMAKE_MODULE_PATH "${vtk_cmake_dir}")
+        #####
+
+        # DELETE any of the third party finders, before we build - see note above
+        remove_files_by_mask(os.path.join(self._source_subfolder,"CMake"), "Find*.cmake")
+
+        # Delete VTK's cmake patches (these support older cmake programs).
+        # We do not have to support old cmake: we require an updated cmake instead.
+        if _support_old_ci_20220514:
+            rmdir(os.path.join(self._source_subfolder,"CMake","patches"))
+        else:
+            rmdir(self, os.path.join(self._source_subfolder,"CMake","patches"))
+
+        # And apply our patches.  I do it here rather than in build, so I can repeatedly call build without applying patches.
+        apply_conandata_patches(self)
+
+
+
+    def source(self):
+        if self.options.use_source_from_git:
+            self.run("git clone -b release --single-branch " + self.git_url + " " + self._source_subfolder)
+            # note: we give the branch a name so we don't have detached heads
+            # TODO change to standard git and python chdir
+            git_hash = "v" + self.version
+            self.run("cd " + self._source_subfolder + " && git checkout -b branch-" + git_hash + " " + git_hash)
+        else:
+            get(**self.conan_data["sources"][self.version],
+                    strip_root=True,
+                    destination=self._source_subfolder,
+                    )
+
+        if self.no_copy_source:
+            self._patch_source()
+
+
+    def _third_party(self):
+        parties = {
+                # LEFT field:  target name for linking, will be used as TARGET::TARGET in package_info()
+                # RIGHT field: package/version to require
+                  "double-conversion": "double-conversion/3.2.0"
+                , "eigen":             "eigen/3.4.0"
+                , "exprtk":            "exprtk/0.0.1"
+                , "fmt":               "fmt/8.1.1"       # 9.1.0 release docs mention a PR - confirmed merged 8.1.0
+                , "freetype":          "freetype/2.11.1" # ... what? ... WORKED with 2022-05-06-freetype
```

Comment:
what is `2022-05-06-freetype`?

Reply:
I think that was a comment by me related to the problems I've had with Freetype in other PRs and issues on conan.
I'm currently using a patched conan, waiting to see if it works without patching with conan 1.48.1 or 1.49
conan-io/conan#11201